lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,41 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          secrets = nil
+          if config.is_a?(String)
+            secrets = [config]
+          else
+            secrets = config.values
+          end
+
+          tokens = {}
+          secrets.each do |secret|
+            key, secret = secret.split('=').map(&:strip)
+            pull_request = self.data.pull_request ? self.data.pull_request : ""
+            now = Time.now.to_i()
+            payload = {"iss" => "travis-ci.org",
+                      "slug" => self.data.slug,
+                      "pull-request" => pull_request,
+                      "exp" => now+5400,
+                      "iat" => now}
+            token = JWT.encode(payload, secret)
```

Comment:
The build that failed initially (https://staging.travis-ci.org/travis-ci/travis_staging_test/builds/507055) had the data encrypted incorrectly, I believe, using the encryption key of a different repository. This results in a stack trace such as this: https://gist.github.com/BanzaiMan/a8e7116c6a80abf34e1fa48efa1e9dcb. (The trace corresponds to https://github.com/travis-ci/travis-build/pull/356/commits/bb16a55b414a22dd66ab41c924251d85a665efff.)